extensions/wikia/AdEngine3/src/slots.js

```diff
@@ -360,21 +354,25 @@ export default {
 		document.addEventListener('scroll', pushSlotAfterComments);
 	},
 
-	injectIncontentPlayer() {
-		const header = document.querySelectorAll('#mw-content-text > h2')[1];
+	/**
+	 * Checks whether ICP can be applied on this page
+	 *
+	 * @returns {boolean}
+	 */
+	isIncontentPlayerApplicable() {
+		const slotName = 'incontent_player';
 
-		if (!header || !isIncontentPlayerApplicable()) {
-			return;
-		}
+		return !context.get('custom.hasFeaturedVideo') && slotInjector.inject(slotName, true);
```

Comment:
What about injecting the container at the beginning and then deciding if it's applicable based on the result of injection? (instead of executing dryRun)